core/sequencer/server.go

```diff
@@ -376,8 +383,8 @@ func (s *Server) ApplyRevision(ctx context.Context, in *spb.ApplyRevisionRequest
 	defer func() { fnLatency.Observe(time.Since(start).Seconds(), in.DirectoryId, "ApplyRevision") }()
 	meta, err := s.batcher.ReadBatch(ctx, in.DirectoryId, in.Revision)
 	fnLatency.Observe(time.Since(start).Seconds(), in.DirectoryId, "ReadBatch")
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "ReadBatch(%v, %v): %v", in.DirectoryId, in.Revision, err)
+	if st := status.Convert(err); st.Code() != codes.OK {
```

Comment:
Could you add a comment saying that it's important to preserve the `NotFound` error here if it occurs? Also, double check that `ReadBatch` has the corresponding `NotFound` note in the interface definition.

Reply:
I've added the NotFound error code case into the proto API definition.